packages/edit-site/src/components/sidebar/index.js

```diff
@@ -1,117 +1,81 @@
+	const [ { navDirection, focusSelector }, setNavDirection ] = useState( {
+		navDirection: null,
+		focusSelector: null,
+	} );
 
-	return (
-		<>
-			<SidebarScreenWrapper path="/">
-				<SidebarNavigationScreenMain />
-			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/navigation">
-				<SidebarNavigationScreenNavigationMenus />
-			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/navigation/:postType/:postId">
-				<SidebarNavigationScreenNavigationMenu />
-			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/wp_global_styles">
-				<SidebarNavigationScreenGlobalStyles />
-			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/page">
-				<SidebarNavigationScreen
-					title={ __( 'Manage pages' ) }
-					content={ <DataViewsSidebarContent /> }
-				/>
-			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/page/:postId">
-				<SidebarNavigationScreenPage />
-			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/:postType(wp_template)">
-				<SidebarNavigationScreenTemplatesBrowse />
-			</SidebarScreenWrapper>
-			{ ! isMobileViewport && (
-				<SidebarScreenWrapper path="/patterns">
-					<SidebarNavigationScreenPatterns />
-				</SidebarScreenWrapper>
-			) }
-			<SidebarScreenWrapper path="/:postType(wp_template_part)/all">
-				<SidebarNavigationScreenTemplatesBrowse />
-			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/:postType(wp_template_part|wp_block)/:postId">
-				<SidebarNavigationScreenPattern />
-			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/:postType(wp_template)/:postId">
-				<SidebarNavigationScreenTemplate />
-			</SidebarScreenWrapper>
-		</>
-	);
-}
+	const navigate = useCallback( ( direction, backFocusSelector ) => {
+		setNavDirection( ( prevDir ) => ( {
+			navDirection: direction,
+			focusSelector:
+				direction === 'forward' && backFocusSelector
+					? backFocusSelector
+					: prevDir.focusSelector,
+		} ) );
+	}, [] );
+
+	const wrapperRef = useRef();
+	useEffect( () => {
+		let elementToFocus;
+		if ( navDirection === 'forward' ) {
+			const [ firstTabbable ] = focus.tabbable.find( wrapperRef.current );
+			elementToFocus = firstTabbable ?? wrapperRef.current;
+		} else if ( navDirection === 'back' && focusSelector ) {
+			elementToFocus = wrapperRef.current.querySelector( focusSelector );
+		}
+		elementToFocus?.focus();
+	}, [ navDirection, focusSelector ] );
 
-function Sidebar() {
-	const { params: urlParams } = useLocation();
-	const initialPath = useRef( getPathFromURL( urlParams ) );
+	const disableMotion = useReducedMotion();
+	const { initial, animate } = getAnim( navDirection );
 
 	return (
-		<>
-			<NavigatorProvider
-				className="edit-site-sidebar__content"
-				initialPath={ initialPath.current }
-			>
-				<SidebarScreens />
-			</NavigatorProvider>
-			<SaveHub />
-		</>
+		<SidebarNavigationContext.Provider value={ navigate }>
+			<div className="edit-site-sidebar__content">
+				<motion.div
```

Comment:
Unfortunately, this animation ends up a bit clunky even on my high-end machine. 

@ciampo refactored this animation to CSS in https://github.com/WordPress/gutenberg/pull/56909 to avoid a visible performance glitch reported in https://github.com/WordPress/gutenberg/issues/55892. The root cause was never fixed, but the CSS animations were resolving the clunkyness, at least.

Before this PR:


https://github.com/WordPress/gutenberg/assets/8436925/23be4c3d-85b3-47b0-aea8-cbd7f78c60a3

After this PR:

https://github.com/WordPress/gutenberg/assets/8436925/7ed7406e-6b90-4a0c-bb4b-3da1a7536511